src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWING_COEFFICIENT = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            int newLength = (int) (elementData.length * GROWING_COEFFICIENT);
+            resize(newLength);
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (size == elementData.length) {
+            int newLength = (int) (elementData.length * GROWING_COEFFICIENT);
+            resize(newLength);
+        }
+        verifyIndex(index);
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        set(value, index);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        verifyIndex(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        verifyIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        verifyIndex(index);
+        T object = get(index);
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        size--;
+        return object;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (elementsAreEqual(element, elementData[i])) {
+                remove(i);
+                return element;
+            }
+        }
+        throw new NoSuchElementException("No such item found: " + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void resize(int newLength) {
+        if (newLength < size) {
+            throw new IllegalArgumentException("New length is less than the current size");
+        }
+        T[] newArray = (T[]) new Object[newLength];
```

Comment:
This method should be private and then you don't need to check if (newLength < size). But you can leave that if in. It's not a mistake.